KeenClient/KIODBStore.m

```diff
@@ -50,17 +54,27 @@ @implementation KIODBStore {
 - (instancetype)init {
     self = [super init];
 
-    if(self) {
+    if (nil != self) {
```

Comment:
Seems that it's mostly a stylistic preference in ObjC land, and for whatever reason doesn't seem consistent in code I've seen. We can decide to formalize a style if you'd like.

The `init` code snippet from Xcode looks like this:
```
- (instancetype)init
{
    self = [super init];
    if (self) {
        <#statements#>
    }
    return self;
}
```

I generally like to do a direct comparison in C-like languages because I think it makes the intent clearer, but seems that isn't generally the case in ObjC?

Regarding the actual logic, nil is actually just a preprocessor definition to ((void*)0), and of course 0 is always coerced to false in C, so there aren't any gotchas there.

Logically it's equivalent, so we're splitting hairs about readability here, and that can largely be subjective. I'd vote for generally comparing to nil, but I know we aren't consistently doing that across the project, and other contributors seem to have disagreed with that. All that said, I don't think I'm going to be spending much time making sure we're consistent with this except for cases where it's definitely not clear the intention.